src/agent.ts

```diff
@@ -0,0 +1,139 @@
+/**
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {CpuProfilerConfig, HeapProfilerConfig, ProfileAgentConfig}
+    from './config'
+
+import {AuthenticationConfig, ServiceConfig, Common, Logger, Service} 
+    from '../third_party/types/common-types'
+export const common: Common = require('@google-cloud/common');
+
+
+const pjson = require('../../../package.json');
+const path = require('path');
+const extend = require('extend');
+
+// TODO: finish implementing ProfileAgent
+export class ProfileAgent {
+  // projectId is the Cloud Console project ID to use instead of the one read
+  // from the VM metadata server.
+  //
+  // Set this if running the agent in your local environment or anywhere
+  // outside of Google Cloud Platform.
+  projectId: string;
+
+  // instance is the virtual machine instance to be used instead of the one read
+  // from the VM metadata server.
+  instance?: string;
+
+  // zone is the zone to be used instead of the one read from the VM metadata
+  // server.
+  zone?: string;
+  serviceContext: {
+    // service specified the name of the service under which the profiled data
+    // will be recorded and exposed at the Cloud Profiler UI for the project.
+    // You can specify an arbitray string, but see deployment.target at
+    // https://github.com/googleapis/googleapis/blob/master/google/devtools/cloudprofiler/v2/profiler.proto
+	  // for restrictions.
+    // The string should be the same across different replicas of your service.
+    // so that a globally constant profiling rate is maintained. 
+    service: string;
+
+    // version is a field specifying the version of the service.
+    // It can be an arbitrary string. Cloud Profiler profiles each version of
+    // each service in each zone once per minute.
+    // service defaults to an empty string. 
+    version?: string;
+  };
+  
+  logger: Logger
+  service: Service;
+
+  constructor(config: ProfileAgentConfig) {
+    let initializedConfig: ProfileAgentConfig = ProfileAgent.initConfig(config);
```

Comment:
Here an elsewhere, prefer `const` over `let` if the variable is not going to be redefined. I wonder why the linter is not catching this.